src/SFA.DAS.EmployerFinance.UnitTests/Services/ProviderServiceTests/ProviderServiceFromDb/WhenIGetAProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.EmployerFinance.Data;
+
+namespace SFA.DAS.EmployerFinance.UnitTests.Services.ProviderServiceTests.ProviderServiceFromDb
+{
+    internal class WhenIGetAProvider
+    {
+        private EmployerFinance.Services.ProviderServiceFromDb _sut;
+        private Mock<IDasLevyRepository> _mockDasLevyRepository;
+
+        private string _providerName;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _providerName = Guid.NewGuid().ToString();
+
+            _mockDasLevyRepository = new Mock<IDasLevyRepository>();
+
+            _mockDasLevyRepository
+                .Setup(m => m.FindHistoricalProviderName(It.IsAny<long>()))
+                .ReturnsAsync(new List<string>{_providerName});
+
+            _sut = new EmployerFinance.Services.ProviderServiceFromDb(_mockDasLevyRepository.Object);
+        }
+
+        [Test]
+        public async Task ThenTheHistoricalProviderNameIsRetrievedFromTheRepository()
+        {
+            // arrange 
+            long ukPrn = RandomNumber(1, 100);           
```

Comment:
Just a quick note rather than a change required. Unless randoms values are required to properly test some logic its better to use defined values as it ensures the tests are always consistent. The problem with using random values is a test could pass one run and fail the next purely based in the values passed. What we really want is the test to fail constantly if the code is incorrect.

Reply:
Actually I'd recommend removing the random number generation unless needed. Worth checking with the rest of the team though to see if everyone else agrees and wishes to go use that approach.

Reply:
Removed random number generation